ios/browser/api/ads/brave_ads.mm

```diff
@@ -357,6 +362,8 @@ - (BOOL)isEnabled {
 
 - (void)setEnabled:(BOOL)enabled {
   self.prefs[kEnabledPrefKey] = @(enabled);
+  self.prefs[kBraveNewsOptedInPrefKey] = @(enabled);
```

Comment:
You probably need to set these prefs on init based on the enabled pref if it's not set otherwise existing users will hit the CHECK's still